arcade/draw_commands.py

```diff
@@ -588,18 +579,18 @@ def draw_polygon_outline(point_list: PointList,
     new_point_list.append(point_list[0])
 
     triangle_point_list = []
-    # This needs a lot of improvement
-    last_point = None
-    for point in new_point_list:
-        if last_point is not None:
-            points = get_points_for_thick_line(last_point[0], last_point[1], point[0], point[1], line_width)
-            reordered_points = points[1], points[0], points[2], points[3]
-            triangle_point_list.extend(reordered_points)
-        last_point = point
-
-    points = get_points_for_thick_line(new_point_list[0][0], new_point_list[0][1], new_point_list[1][0],
-                                       new_point_list[1][1], line_width)
-    triangle_point_list.append(points[1])
+    for i in range(len(point_list) - 1):
+        point1 = point_list[i]
+        point2 = point_list[i + 1]
+        
+        points = get_points_for_thick_line(point1[0], point1[1], point2[0], point2[1], line_width)
+        triangle_point_list.extend([points[1], points[0], points[2], points[3]])
+    
+    # Handle the last segment (connecting back to the start)
+    point1 = point_list[-1]
+    point2 = point_list[0]
+    points = get_points_for_thick_line(point1[0], point1[1], point2[0], point2[1], line_width)
+    triangle_point_list.extend([points[1], points[0], points[2], points[3]])
```

Comment:
1. Why is this in a type hint PR?
2. Do we have unit tests covering these changes?

If this is a needed improvement. If it works, this should be in another, separate PR.